### PR TITLE
Feature/explorer edit item

### PR DESCRIPTION
**Pull request by peratik**
## Changes
- Edit item placeholder added
- Actions refactored
- Icons removed https://discord.com/channels/334289660698427392/968805941824876544/968808371400638514

## Checklist

- [x] I've performed a self-review, followed all rules from [Frontend style guide](https://www.notion.so/santiment/Front-end-style-guide-81750096b38c4bea9a29b14fd4ab8667)
- [x] If I make changes to another person's module, I've asked how to use it or request a review
- [x] I've updated the [documentation](https://github.com/santiment/academy), if necessary (Keyboard shortcuts, Account settings)
- [x] I've checked night mode, mobile & tablet screens (if have changes in UI)
- [x] I've added tests (if necessary)

## Screenshots or GIFs

image

### CODE REVIEW

File: src/pages/Explorer/Components/Actions.svelte [hunk 1]
```diff
@@ -64,19 +62,33 @@
...
+
+  function onEdit(e) {
+    e.preventDefault()
+    showEditDialog({
+      id,
+      item,
+      singular,
+      title: item.trigger ? item.trigger.title : item.title,
+      description: item.trigger ? item.trigger.description : item.description,
+      isPublic: item.trigger ? item.trigger.isPublic : item.isPublic,
```
Comment (DmitriVanGuard) [inline src/pages/Explorer/Components/Actions.svelte:81]: Destruct necessary fields above:
```js
const {title, description, isPublic} = item.trigger || item
```

File: src/pages/Explorer/Components/EditDialog.svelte [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+  const MAX_TITLE_LENGTH = 25
+  const MIN_TITLE_LENGTH = 3
+
+  export let itemData
+
+  let closeDialog
+  let editedTitle = itemData.title
+  let editedDescription = itemData.description ? itemData.description : ''
+  let editedPublic = !!itemData.isPublic
```
Comment (DmitriVanGuard) [inline src/pages/Explorer/Components/EditDialog.svelte]:
```suggestion
  $: ({title, description, isPublic} = item)
```

File: src/pages/Explorer/Components/EditDialog.svelte [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+  const MAX_TITLE_LENGTH = 25
+  const MIN_TITLE_LENGTH = 3
+
+  export let itemData
+
+  let closeDialog
+  let editedTitle = itemData.title
+  let editedDescription = itemData.description || ''
+  let editedPublic = !!itemData.isPublic
```
Comment (DmitriVanGuard) [inline src/pages/Explorer/Components/EditDialog.svelte:23]:
```suggestion
  $: ({id, editKey, title, description, isPublic} = item)
```

File: src/pages/Explorer/Components/EditDialog.svelte [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  let loading = false
+
+  $: titleLength = editedTitle ? editedTitle.length : 0
```
Comment (DmitriVanGuard) [inline src/pages/Explorer/Components/EditDialog.svelte:26]:
```suggestion
  $: titleLength = (title || '').length
```

**Review by DmitriVanGuard — CHANGES_REQUESTED** [review]

File: src/pages/Explorer/Components/EditDialog.svelte (hunk 3 above)
Comment (peratik) [reply src/pages/Explorer/Components/EditDialog.svelte:23]: there is `item` property in `itemData` that we need to use that later, we cant rename it

File: src/pages/Explorer/Components/EditDialog.svelte [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+<script>
+  import Dialog from 'webkit/ui/Dialog'
+  import Toggle from 'webkit/ui/Toggle'
+  import { edit } from './api'
+
+  const MAX_TITLE_LENGTH = 25
+  const MIN_TITLE_LENGTH = 3
+
+  export let itemData
```
[the remark this answers is not in the document]
Comment (peratik) [reply src/pages/Explorer/Components/EditDialog.svelte:18]: there is item property in itemData that we need to use that later, we cant rename it

File: src/pages/Explorer/Components/EditDialog.svelte (hunk 3 above)
Comment (peratik) [reply src/pages/Explorer/Components/EditDialog.svelte:24]: it does not allow me to edit values in the form

File: src/pages/Explorer/Components/EditDialog.svelte (hunk 5 above)
[the remark this answers is not in the document]
Comment (DmitriVanGuard) [reply src/pages/Explorer/Components/EditDialog.svelte:18]: Makes sense

File: src/pages/Explorer/Components/EditDialog.svelte (hunk 3 above)
Comment (DmitriVanGuard) [reply src/pages/Explorer/Components/EditDialog.svelte:24]: Sorry, it should be a simple `let` statement, not a reactive one

**Review by DmitriVanGuard — APPROVED** [review]

**Pull request merged**